tests/bot/cogs/test_tags.py

```diff
@@ -0,0 +1,305 @@
+                self.assertTrue("description" in v["embed"])
+
+    @patch("bot.cogs.tags.REGEX_NON_ALPHABET")
+    def test_fuzzy_search(self, regex):
+        """Should return correct words match rate."""
+        test_cases = [
+            {
+                "args": ("foo", "foo"),
+                "expected_output": 100.0,
+                "regex_split": iter(["foo"])
+            },
+            {
+                "args": ("foo", "bar"),
+                "expected_output": 0.0,
+                "regex_split": iter(["bar"])
+            },
+            {
+                "args": ("foo", "fuu"),
+                "expected_output": 33.33333333333333,
+                "regex_split": iter(["fuu"])
+            }
+        ]
+
+        for case in test_cases:
+            with self.subTest(f"Should return {case['expected_output']} as match rate.", args=case["args"]):
+                regex.sub.return_value = "foo"
+                regex.split.return_value = case["regex_split"]
+
+                actual = tags.Tags._fuzzy_search(*case["args"])
+
+                self.assertEqual(actual, case["expected_output"])
+                regex.sub.called_once_with("", case["args"][0].lower())
+                regex.split.called_once_with(case["args"][1].lower())
+
+    def test_get_tag(self):
+        """Should return correct tag from Cog cache."""
+        cache = self.cog._cache
+        test_cases = [
+            {"name": tag_name, "expected": tag} for tag_name, tag in cache.items()
+        ]
+        test_cases.append(
+            {
+                "name": "retur",
+                "expected": cache["return"]
+            }
+        )
+        for case in test_cases:
+            with self.subTest(tag_name=case["name"], expected=case["expected"]):
+                actual = self.cog._get_tag(case["name"])
+                self.assertEqual(actual[0], case["expected"])
+
+    def test_get_suggestions(self):
+        """Should return correct list of tags and interact with `_fuzzy_search`."""
+        cache = self.cog._cache
+        test_cases = [
+            {
+                "args": ("codeblck", None),
+                "expected": [cache["codeblock"]]
+            },
+            {
+                "args": ("pep", None),
+                "expected": [cache["pep8"]]
+            },
+            {
+                "args": ("class", None),
+                "expected": [cache["classmethod"], cache["class"]]
+            },
+            {
+                "args": ("o-topc", [100, 90, 80, 70, 60, 50, 40, 30, 20, 10]),
+                "expected": [cache["off-topic"]]
+            },
+            {
+                "args": ("my-test-string", None),
+                "expected": []
+            }
+        ]
+        for case in test_cases:
+            with self.subTest(args=case["args"], expected=case["expected"]):
+                actual = self.cog._get_suggestions(*case["args"])
+
+                self.assertEqual(actual, case["expected"])
```

Comment:
Is there any reason to compare the full contents of the tag rather than just the tag name (dictionary key) in this test?